### PR TITLE
Add comments feature and navigation bar with parallax scrolling

### PR DESCRIPTION
**Pull request by rtnavarr**
- Created a Comment class in Comment.java and modified DataServlet.java to store comments entered in an html form. 
- Modified script.js to fetch JSON/parse an array of comments from the /data server.
- Deleted hard-coded message display from the previous iteration of the week 3 walkthrough. 
- Modified index.html to display submitted comments in an unordered list.
- Created a navigation bar and added parallax and smooth scrolling.
- Updated comments and resolved merge conflicts.

### CODE REVIEW

File: rachelnavarro/step/portfolio/src/main/java/com/google/sps/data/Comment.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+/** Class representing a comment submitted by a user. */
+public class Comment {
+
+  private final String body;
+  
+  private final String name;
+
+  public Comment(String userBody, String userName) {
+    this.body = userBody;
+    this.name = userName;
```
Comment (dchin2) [inline rachelnavarro/step/portfolio/src/main/java/com/google/sps/data/Comment.java]: I would make these variable names the same, either userName or name.

File: rachelnavarro/step/portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -24,14 +25,28 @@ function drawCardOfTheDay() {
...
-function displayGreeting() {
-  fetch('/data').then(response => response.text()).then((greeting) => {
-    document.getElementById('greeting-container').innerText = greeting;
+/** Fetches and loads all previous comments in an unordered list. */
+function displayComments() {
+  fetch('/data').then(response => response.json()).then((allcomments) => {
+    var i;
+    for (i = 0; i < allcomments.length; i++) {
+      var datetime = getDateTime();
+      createCommentEntry(allcomments[i].name + ' said "' + allcomments[i].body + "' at " + datetime);
```
Comment (dchin2) [inline rachelnavarro/step/portfolio/src/main/webapp/script.js]: Are the quotes here correct?

I would use a format string, https://stackoverflow.com/questions/610406/javascript-equivalent-to-printf-string-format/32202320#32202320

It is easier to read :)

Something like \`${allComments[i].name} said "${allComments[i].body}" ... etc\`

Also rename allcomments to allComments.

File: rachelnavarro/step/portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -24,14 +25,28 @@ function drawCardOfTheDay() {
...
+  var node = document.createElement("LI");
+  var textnode = document.createTextNode(text);
+  node.appendChild(textnode);
+  document.getElementById("comments-container").appendChild(node);
+}
+
+/** Gets date/time in YYYY-MM-DD HH:MM:SS format. */
+function getDateTime() {
+  var today = new Date();
+  var datetime = today.getFullYear() + '-' + (today.getMonth()+1) + '-' + today.getDate() + " " + today.getHours() + ":" + today.getMinutes() + ":" + today.getSeconds();
```
Comment (dchin2) [inline rachelnavarro/step/portfolio/src/main/webapp/script.js]: same here with format string

**Review by dchin2 — CHANGES_REQUESTED** [review]

File: rachelnavarro/step/portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (rtnavarr) [reply rachelnavarro/step/portfolio/src/main/webapp/script.js]: Fixed! @dchin2

File: rachelnavarro/step/portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -27,11 +27,12 @@ function drawCardOfTheDay() {
 
 /** Fetches and loads all previous comments in an unordered list. */
 function displayComments() {
-  fetch('/data').then(response => response.json()).then((allcomments) => {
+  fetch('/data').then(response => response.json()).then((allComments) => {
     var i;
-    for (i = 0; i < allcomments.length; i++) {
+    for (i = 0; i < allComments.length; i++) {
       var datetime = getDateTime();
```
Comment (dchin2) [inline rachelnavarro/step/portfolio/src/main/webapp/script.js:33]: change to dateTime, here and other places.

**Review by dchin2 — APPROVED** [review]
lgtm after changes

**Pull request merged**